.github/workflows/master_dev_checks.yml

```diff
@@ -7,310 +7,75 @@
 
     dev-checks:
       runs-on: non-build
+      env:
+        CI-BRANCH: TDEVOPS-1832
+        BASE_REF: "${{github.base_ref}}"
+        BASE_SHA: ${{ github.event.pull_request.base.sha }}
+        GITHUB_EVENT_BEFORE: "${{ github.event.before }}"
+        GITHUB_EVENT_NAME: "${{ github.event_name }}"
+        GITHUB_SHA: "${{ github.sha }}"
+        GITHUB_TOKEN: ${{ secrets.CIPIPELINE_GITHUB_TOKEN }}
+        HEAD_BRANCH: ${{ github.event.pull_request.head.ref}}
+        HEAD_SHA: ${{ github.event.pull_request.head.sha }}
+        JIRA_USERNAME: ${{ secrets.JIRA_USERNAME }}
+        JIRA_API_TOKEN: ${{ secrets.JIRA_API_TOKEN }}
+        OWNER: "${{ github.repository_owner }}"
+        PR_TITLE: "${{github.event.pull_request.title}}"
+        REPO: "${{  github.repository }}"
+        SUPPORTED_JIRA_PROJECTS: "${{vars.SUPPORTED_JIRA_PROJECTS}}" 
+        USER: "${{github.event.pull_request.user.login}}"
       steps:
+      
+        - uses: actions/checkout@v4.1.1
+          with:
+            fetch-depth: 0
+
         - name: Dump GitHub context
           env:
             GITHUB_CONTEXT: ${{ toJson(github) }}
           run: |
             echo "$GITHUB_CONTEXT"
 
-        - uses: actions/checkout@v4.1.1
-          with:
-            fetch-depth: 0
-
-        - name: Get branch name
-          shell: bash
+        - name: Setup CI Scripts
           run: |
-            echo "SOURCE_BRANCH=$(echo ${{ github.event.pull_request.head.ref }} | tr / -)" >> $GITHUB_ENV
-            echo "TARGET_BRANCH=$(echo ${GITHUB_BASE_REF} | tr / -)" >> $GITHUB_ENV
-            echo "GITHUB_TOKEN=${{ secrets.CIPIPELINE_GITHUB_TOKEN  }}" >> $GITHUB_ENV
+            cd ~/convoy-ci
+            git checkout ${{env.CI-BRANCH}}
```

Comment:
Add `git fetch` before doing `git checkout`; Othewise the `git checkout` will fail for newly created branches. Also add `set +e` to fail early. Otherwise any failures here will go undetected.